### PR TITLE
HAWQ-1187. Fix "-Wsometimes-uninitialized, -Wsizeof-pointer-memaccess , -Wdangling-else, -Wheader-guard, -Wenum-conversion, -Winvalid-source-encoding" compile warnings under osx.

### PR DESCRIPTION
**Pull request by xunzhang**
http://pastebin.com/VWSuCwsC

### CODE REVIEW

**Comment by xunzhang** [comment]
cc @paul-guo- @ictmalili

**Comment by xunzhang** [comment]
also cc @stanlyxiang

File: src/bin/pg_dump/dumputils.c [hunk 1]
```diff
@@ -1062,7 +1062,7 @@ char *
 custom_fmtopts_string(const char *src)
 {
 		int			len = src ? strlen(src) : 0;
-		char	   *result = malloc(len * 2 + 1);
+		char	   *result = calloc(1, len * 2 + 1);
```
Comment (ictmalili) [inline src/bin/pg_dump/dumputils.c]: I think this is a weird usage, for the usage of calloc, the first parameter should be count, and the second should be size. What about call(len*2+1, sizeof(char)) ?

**Comment by ictmalili** [comment]
LGTM now. +1

**Comment by paul-guo-** [comment]
+1

**Comment by stanlyxiang** [comment]
＋1

**Comment by xunzhang** [comment]
Merged into master.

**Pull request closed without merging**